src/sfloader/fluid_sfont.c

```diff
@@ -700,13 +700,40 @@ int fluid_sample_set_pitch(fluid_sample_t *sample, int root_key, int fine_tune)
  */
 int fluid_sample_validate(fluid_sample_t *sample, unsigned int buffer_size)
 {
+#define EXCLUSIVE_FLAGS (FLUID_SAMPLETYPE_MONO | FLUID_SAMPLETYPE_RIGHT | FLUID_SAMPLETYPE_LEFT)
+    static const unsigned int supported_flags = EXCLUSIVE_FLAGS | FLUID_SAMPLETYPE_LINKED | FLUID_SAMPLETYPE_OGG_VORBIS | FLUID_SAMPLETYPE_ROM;
+
     /* ROM samples are unusable for us by definition */
     if(sample->sampletype & FLUID_SAMPLETYPE_ROM)
     {
         FLUID_LOG(FLUID_WARN, "Sample '%s': ROM sample ignored", sample->name);
         return FLUID_FAILED;
     }
 
+    if(sample->sampletype & ~supported_flags)
+    {
+        FLUID_LOG(FLUID_WARN, "Sample '%s' has unknown flags, which might indicate that it uses a compression that it not supported by fluidsynth", sample->name);
```

Comment:
...it *is* not supported...

Reply:
Actually, I've already seen a similar error message from fluidsynth when I tried to load a FLAC compressed soundfont with a libsndfile that was lacking FLAC support. So we're checking the flag, but not if the compressed data is actually OGG/Vorbis but simply accept all formats that libsndfile understands. Quite handy for my tests... but maybe not desired?

Reply:
> So we're checking the flag, but not if the compressed data is actually OGG/Vorbis but simply accept all formats that libsndfile understands.

I do not quite understand this sentence, sorry. Anyway:

* If the sample is FLAC, it (should) have the 0x20 flag set. In this case, I would expect fluidsynth to treat it's content as RAW PCM, which would probably sound very glitchy.
* If the sample is FLAC and (incorrectly) has the 0x10 flag (OGG) set, it would successfully be decoded by libsndfile, yes (assuming it was compiled with FLAC support).

But in any case, I would vote to drop the sample if we detect unknown flags. Until we decide to support FLAC compressed samples... if we decide. (_I'm going offline now_)

Reply:
>> So we're checking the flag, but not if the compressed data is actually OGG/Vorbis but simply accept all formats that libsndfile understands.
> I do not quite understand this sentence, sorry.

We check the sample type flag for valid values, like 0x10 for OGG. And then we simply pass the sample data on to libsndfile to be decoded. But we leave it up to libsndfile to determine the data format, and therefore accept all audio data that libsndfile can understand. Doesn't seem like it can cause harm, but as you propose to add warnings about inconsistencies in the Soundfont in this PR, I though it might be another place for a possible warning.

> Until we decide to support FLAC compressed samples...

We effectively already support FLAC compressed samples :-)

Reply:
> We effectively already support FLAC compressed samples :-)

Well... yes, but we should not. Neither have we checked, whether libsndfile has been compiled with FLAC support, nor are we checking the 0x20 flag anywhere. A sample having 0x10 set and containing anything but OGG is broken and IMO should not be used.

> but as you propose to add warnings about inconsistencies in the Soundfont in this PR, I though it might be another place for a possible warning.

Ok, so you want warn if `SF_INFO::format` is something else than `SF_FORMAT_OGG`. Only warn or also reject the sample? (I would vote for rejecting them, to avoid people using FLAC samples via a backdoor in fluidsynth.)

Reply:
Sorry about the late reply!

> Ok, so you want to warn if SF_INFO::format is something else than SF_FORMAT_OGG. Only warn or also reject the sample? (I would vote for rejecting them, to avoid people using FLAC samples via a backdoor in fluidsynth.)

After thinking about this more, maybe simply leave it as it is? To check if the sample data is really OGG encoded, we would need to add extra code. And there is really no pressing need to guard against this case. One, because libsndfile will check and throw errors if it can't parse the data. And second, because we're a synth and not a soundfont validator.

Reply:
> After thinking about this more, maybe simply leave it as it is? To check if the sample data is really OGG encoded, we would need to add extra code. And there is really no pressing need to guard against this case. One, because libsndfile will check and throw errors if it can't parse the data.

I don't see a pressing need either. It's just from my understanding of the (non-existent) SF3 spec, that samples marked as OGG should also contain OGG. Otherwise, we would support FLAC, opus, whatever via a backdoor. A backdoor only being available due to an implementation detail (because we've chosen libsndfile, which can cope with it somehow). So, if we are not rejecting it, I would at least warn, to indicate the broken- and unportableness of that soundfont.

> And second, because we're a synth and not a soundfont validator.

Pls. keep in mind that fluidsynth also checks for the correct order of instrument/preset generators. Doesn't really make any sense, but is required by the spec.